Documentation (#358)
aramis-lab/leaspy: docs/algorithms.md

### Setting Options (Different Models)
   - How to set specific options for different types of models.
   - Customizing parameters for logistic, joint, mixture, and covariate models.

## Personalize

The idea of this section is to describe how the inference over new unseen patients could be done. For this random effects of new patients should be estimated using some follow-up visits to be able to describe their progression and make predictions about their future progression. Note here that we assume that fixed effects have already been estimated and are fixed. Two main approach for estimating the random effects exist in Leaspy. 

__ more of a frequentist one:__ random effects are estimated using the solver [_minimise_](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html) from the package Scipy (\cite{2020SciPy_NMeth}) to maximize the likelihood knowing the fixed effects.

```python

> >>> personalize_settings = AlgorithmSettings("scipy_minimize", seed=0)

Author: It's a copy-paste from the nut-shell code, is it an issue? Should I make a reference to it instead of copy-pasting? I have copy-pasted because I found it clearer for the reader and more coherent with the following section. WDYT @caglayantuna and @NicolasGensollen ?

Reviewer: Hi @JulietteOrtholand 

You mean that you copy-pasted the python code from another page and "just" want to include it here through a reference, right ?

Reviewer: I think you can copy-paste for now. There are different ways to avoid that copy-pasting, but none doing exactly what we need here (at least in what I took a look at). Once we have a better vision of the full documentation, we could take the time to optimize this kind of stuff.

Author: Thanks!

>>> algorithm = algorithm_factory(personalize_settings)
>>> individual_parameters = algorithm.run(model, dataset)
==> Setting seed to 0
|##################################################|   200/200 subjects
Personalize with `AlgorithmName.PERSONALIZE_SCIPY_MINIMIZE` took: 29s
The standard-deviation of the noise at the end of the AlgorithmType.PERSONALIZE is: 6.85%
>>> individual_parameters.to_dataframe()
        sources_0  sources_1        tau        xi
ID
GS-001   0.027233  -0.423354  76.399582 -0.386364
GS-002  -0.262680   0.665351  75.137474 -0.577525
GS-003   0.409585   0.844824  75.840012  0.102436
GS-004   0.195366   0.056577  78.110085  0.433171
GS-005   1.424637   1.054663  84.183098 -0.051317
...           ...        ...        ...       ...
GS-196   0.961941   0.389468  72.528786  0.354126
GS-197  -0.561685  -0.720041  79.006042 -0.624100
GS-198  -0.061995   0.177671  83.596138  0.201686
GS-199   1.932454   1.820023  92.275978 -0.136224
GS-200   1.152407  -0.171888  76.504517  0.770118

[200 rows x 4 columns]
```

> __ more a bayesian one:__ random effects are estimated using a Gibbs sampler with an option on the burn-in phase [REF part?] and temperature scheme [REF part?]. Nevertheless, for now the package enable only to extract the mean or the mode of the posterior distribution. 

Author: @KaisaridiSofia Can you confirm you will explain temperature and burn-in in fit part so I can refer to this part here?

Reviewer 2: Yes @JulietteOrtholand, I will do that

### Setting Options (Different Models)
   - How to set specific options for different types of models.
   - Customizing parameters for logistic, joint, mixture, and covariate models.

## Personalize

> The idea of this section is to describe how the inference over new unseen patients could be done. For this random effects of new patients should be estimated using some follow-up visits to be able to describe their progression and make predictions about their future progression. Note here that we assume that fixed effects have already been estimated and are fixed. Two main approach for estimating the random effects exist in Leaspy. 

Author: I kept new

> The idea of this section is to describe how the inference over new unseen patients could be done. For this random effects of new patients should be estimated using some follow-up visits to be able to describe their progression and make predictions about their future progression. Note here that we assume that fixed effects have already been estimated and are fixed. Two main approach for estimating the random effects exist in Leaspy. 

Author: Can I close this thread? 0:)

> The idea of this section is to describe how the inference over new unseen patients could be done. For this random effects of new patients should be estimated using some follow-up visits to be able to describe their progression and make predictions about their future progression. Note here that we assume that fixed effects have already been estimated and are fixed. Two main approach for estimating the random effects exist in Leaspy. 

Author: Discussed, I close the thread

```python
>>> personalize_settings = AlgorithmSettings("scipy_minimize", seed=0)
>>> algorithm = algorithm_factory(personalize_settings)
>>> individual_parameters = algorithm.run(model, dataset)
==> Setting seed to 0
|##################################################|   200/200 subjects
Personalize with `AlgorithmName.PERSONALIZE_SCIPY_MINIMIZE` took: 29s
The standard-deviation of the noise at the end of the AlgorithmType.PERSONALIZE is: 6.85%
>>> individual_parameters.to_dataframe()
        sources_0  sources_1        tau        xi
ID
GS-001   0.027233  -0.423354  76.399582 -0.386364
GS-002  -0.262680   0.665351  75.137474 -0.577525
GS-003   0.409585   0.844824  75.840012  0.102436
GS-004   0.195366   0.056577  78.110085  0.433171
GS-005   1.424637   1.054663  84.183098 -0.051317
...           ...        ...        ...       ...
GS-196   0.961941   0.389468  72.528786  0.354126
GS-197  -0.561685  -0.720041  79.006042 -0.624100
GS-198  -0.061995   0.177671  83.596138  0.201686
GS-199   1.932454   1.820023  92.275978 -0.136224
GS-200   1.152407  -0.171888  76.504517  0.770118

[200 rows x 4 columns]
```

> __more a bayesian one:__ random effects are estimated using a Gibbs sampler with an option on the burn-in phase (see [fit description](##Fit))and temperature scheme [fit description](##Fit). Nevertheless, for now the package enable only to extract the mean or the mode of the posterior distribution. They can be used with the same procedure using `mean_posterior` or `mode_posterior` flag. 

Reviewer 3: ```suggestion
__more a bayesian one:__ random effects are estimated using a Gibbs sampler with an option on the burn-in phase (see [fit description](##Fit))and temperature scheme [fit description](##Fit). Currently, the package enables to extract the mean or the mode of the posterior distribution. They can be used with the same procedure using `mean_posterior` or `mode_posterior` flag. 
```

- outliers
- not enough patients 
- parameters don't converge 

> - score don't progress

Author: It is work in progress I think it should be addressed in another PR
